### DISCUSSION TITLE
How to scrape custom resource labels that are added while Azure resource provisioning (eg:- Project Name, Team Name, App Name etc.)

### DISCUSSION THREAD

**Discussion by sujith-r** [tomkerkhove/promitor#1694] [Q&A] [answered]
Hi Tom,

Can we scrape custom labels which we add to Azure resources at the time of provisioning of resources. As of now the Built in labels like subscription id, rsg, resource uri etc. are being scraped. What about the labels like Project Name, Product Name, Team Name etc. which we add to resources as part of our resource provisioning automation.

Use case :- 

In a larger project team there may be n numbe of sub-teams who own one or more functional components and they may be using specific PaaS resources. Some of the PaaS resources may be shared across teams as well though. To understand resource belongs to which team we make use of the above mentioned labels. For eg:- We have larger products (like platforms) which consists of 10+ teams. This implies within that Project's RSG we may have resources belonging to multiple sub teams working for the project, in a borader perspective we wanted the demarcation or boundary to be these sub teams.

We noted the custom label feature in Promitor but this is something we add to the Promitor configuration rather than the custom labels automatically flowing from the actual resource tag, right?

Thanks,
Sujith

**Comment by tomkerkhove** [comment, marked as answer]
> What about the labels like Project Name, Product Name, Team Name etc. which we add to resources as part of our resource provisioning automation.

With these labels, are you referring to Azure Tags on the resources or not? If so, then you are looking for https://github.com/tomkerkhove/promitor/issues/599.

**Reply by sujith-r** [reply]
Yes had a look at #599. This is exactly what it is.

**Reply by tomkerkhove** [reply]
Good to hear!

**Reply by avwsolutions** [reply]
Hi @tomkerkhove. First of all, great Product and new website :) .  I have a question regarding this  feature. Does this include azure metadata like custom added tags in azure (like owner: SRE or resolver:backoffice) that become prometheus labels ? We really are looking forward for such to do better routing of databases to multiple teams.  I understand #599 is part of the next coming discovery (0.6) release ?

**Reply by avwsolutions** [reply]
If you need some help, let me know. I'm not that good with dotnet, but always ready to support.

**Reply by tomkerkhove** [reply]
This is definitely coming and will surface your custom tags on the resource. What it will look like, however, is not written in stone yet given Prometheus requires us to report the same label count for the same metric.

Happy to hear you like Promitor! If you are using it in production and I can list you, feel free to let me know on kerkhove.tom@gmail.com.

**Reply by avwsolutions** [reply]
Hi @tomkerkhove  send you an e-mail about this.

**Reply by tomkerkhove** [reply]
In terms of the tags themselves, since they might be different per resource representing the same metric that might be a problem since the label count must be the same (resource A has 10 tags, B only has 8)

Would it be acceptable if there would be dedicated system metric per Azure resource with just the tags? A bit similar to the upcoming Subscription & resource group metrics: https://docs.promitor.io/operations/#discovery
